lib/manageiq/appliance_console/cli.rb

```diff
@@ -214,16 +220,13 @@ def set_internal_db
       # initdb, relabel log directory for selinux, update configs,
       # start pg, create user, create db update the rails configuration,
       # verify, set up the database with region. activate does it all!
-      unless config.activate
-        say "Failed to configure internal database"
-        return
-      end
+      raise CliError, "Failed to configure internal database" unless config.activate
 
       # enable/start related services
       config.post_activation
     rescue RuntimeError => e
       say e.message
```

Comment:
I'm pretty sure you can remove this line

and up in the top level rescue add:

```ruby
say e.cause.message if e.cause
```

Reply:
@kbrock Maybe I don't understand what you are suggesting but I what I `think` you are suggesting is to remove the `raise CliError, "Failed to configure internal database" unless config.activate` that will not work.

If a `RuntimeError` is raised this line will not run but if a `RuntimeError` is not raised having this line is important as it ensure the next line `config.post_activation` does not run.

Please explain.

Reply:
Ah. My bad. I see now that you are suggesting to eliminate one of the invocations to `say()`. Will do. Thank you.

Reply:
darn - I could have sworn `cause` existed